app/controllers/objects_controller.rb

```diff
@@ -12,13 +12,28 @@ def show
     render json: PreservedObject.find_by!(druid: druid).to_json
   end
 
+  # return a specific file from the Moab
+  # GET /objects/:druid/file?type=manifest&filepath=signatureCatalog.xml
+  def file
+    err_msgs = file_params_errors(params)
```

Comment:
I know this is a janky way to validate params.  I looked for documentation on how to use ActiveRecord validations in a controller and I kept finding the rails guide admonition repeated in other places about "don't do this in controllers, it gets unwieldy and too hard to maintain" so I gave up and rolled my own.